React/Modules/RCTDevSettings.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <React/RCTBridge.h>
+
+/**
+ * An abstraction for a key-value store to manage RCTDevSettings behavior.
+ * The default implementation persists settings using NSUserDefaults.
+ */
+@protocol RCTDevSettingsDataSource <NSObject>
+
+/**
+ * Updates the setting with the given key to the given value.
+ * How the data source's state changes depends on the implementation.
+ */
+- (void)updateSettingWithValue:(id)value forKey:(NSString *)key;
+
+/**
+ * Returns the value for the setting with the given key.
+ */
+- (id)settingForKey:(NSString *)key;
+
+@end
+
+@interface RCTDevSettings : NSObject
+
+- (instancetype)initWithDataSource:(id<RCTDevSettingsDataSource>)dataSource;
+
+@property (nonatomic, readonly) BOOL isHotLoadingAvailable;
+@property (nonatomic, readonly) BOOL isLiveReloadAvailable;
+@property (nonatomic, readonly) BOOL isRemoteDebugAvailable;
+@property (nonatomic, readonly) BOOL isJSCSamplingProfilerAvailable;
+
+/**
+ * Whether the bridge is connected to a remote JS executor.
+ */
+@property (nonatomic, assign) BOOL isDebuggingRemotely;
+
+/**
+ * Alternate name for the websocket executor, if not the generic term "remote".
+ */
+@property (nonatomic, strong) NSString *websocketExecutorName;
+
+/*
+ * Whether shaking will show RCTDevMenu. The menu is enabled by default if RCT_DEV=1, but
+ * you may wish to disable it so that you can provide your own shake handler.
+ */
+@property (nonatomic, assign) BOOL isShakeToShowDevMenuEnabled;
+
+/**
+ * Whether performance profiling is enabled.
+ */
+@property (nonatomic, assign, setter=setProfilingEnabled:) BOOL isProfilingEnabled;
+
+/**
+ * Whether automatic polling for JS code changes is enabled. Only applicable when
+ * running the app from a server.
+ */
+@property (nonatomic, assign, setter=setLiveReloadEnabled:) BOOL isLiveReloadEnabled;
```

Comment:
Why do we define these properties twice? Once a readonly and once here as writable? I'm actually not sure why they would be writable anyway, since these are all things that are derived from the bridge's execution state.